ui/de-lib/src/main/java/org/iplantc/de/apps/client/presenter/sharing/AppSharingPresenter.java

```diff
@@ -0,0 +1,341 @@
+    public void loadResources() {
+        view.setSelectedApps(selectedApps);
+
+    }
+
+    @Override
+    public void loadPermissions() {
+        permissionsPanel.mask();
+        appService.getPermissions(selectedApps, new LoadPermissionsCallback());
+    }
+
+    @Override
+    public PermissionValue getDefaultPermissions() {
+        return PermissionValue.read;
+    }
+
+    @Override
+    public void processRequest() {
+        JSONObject requestBody = buildSharingJson();
+        JSONObject unshareRequestBody = buildUnSharingJson();
+        if (requestBody != null) {
+            share(requestBody);
+        }
+
+        if (unshareRequestBody != null) {
+            unshare(unshareRequestBody);
+        }
+
+        if (requestBody != null || unshareRequestBody != null) {
+            IplantAnnouncer.getInstance().schedule(appearance.sharingCompleteMsg());
+        }
+
+    }
+
+    private JSONObject buildSharingJson() {
+        JSONObject sharingObj = new JSONObject();
+        FastMap<List<Sharing>> sharingMap = permissionsPanel.getSharingMap();
+
+        if (sharingMap != null && sharingMap.size() > 0) {
+            JSONArray sharingArr = new JSONArray();
+            int index = 0;
+            for (String userName : sharingMap.keySet()) {
+                List<Sharing> shareList = sharingMap.get(userName);
+                JSONObject userObj = new JSONObject();
+                userObj.put("user", new JSONString(userName));
+                userObj.put("apps", buildPathArrWithPermissions(shareList));
+                sharingArr.set(index++, userObj);
+            }
+
+            sharingObj.put("sharing", sharingArr);
+            return sharingObj;
+        } else {
+            return null;
+        }
+    }
+
+    private JSONObject buildUnSharingJson() {
+        JSONObject unsharingObj = new JSONObject();
+        FastMap<List<Sharing>> unSharingMap = permissionsPanel.getUnshareList();
+
+        if (unSharingMap != null && unSharingMap.size() > 0) {
+            JSONArray unsharingArr = new JSONArray();
+            int index = 0;
+            for (String userName : unSharingMap.keySet()) {
+                List<Sharing> shareList = unSharingMap.get(userName);
+                JSONObject userObj = new JSONObject();
+                userObj.put("user", new JSONString(userName));
+                userObj.put("apps", buildPathArr(shareList));
+                unsharingArr.set(index++, userObj);
+            }
+            unsharingObj.put("unsharing", unsharingArr);
+            return unsharingObj;
+        } else {
+            return null;
+        }
+    }
+
+    private JSONArray buildPathArr(List<Sharing> shareList) {
+        JSONArray pathArr = new JSONArray();
+        int index = 0;
+        for (Sharing s : shareList) {
+            pathArr.set(index++, new JSONString(s.getId()));
+        }
+        return pathArr;
+    }
+
+    private JSONArray buildPathArrWithPermissions(List<Sharing> shareList) {
```

Comment:
Similar to the other comment -- these aren't really path-related, they're arrays of apps. So the naming seems confusing.